internal/services/compute/disk_encryption_set_resource.go

```diff
@@ -388,3 +474,15 @@ func diskEncryptionSetRetrieveKeyVault(ctx context.Context, keyVaultsClient *cli
 		softDeleteEnabled:      softDeleteEnabled,
 	}, nil
 }
+
+func diskEncryptionSetValidateRotationPolicy(autoRotate bool, keyVaultKeyID string) error {
+
+	if autoRotate {
+		_, err := keyVaultParse.ParseNestedItemID(keyVaultKeyID)
+		if err == nil {
+			return fmt.Errorf("'auto_key_rotation_enabled' field is set to 'true' expected a key vault key with a versionless ID but version information was found: %q", keyVaultKeyID)
+		}
+	}
+
+	return nil
+}
```

Comment:
I think this will be able to be removed.
```suggestion
```

Reply:
Fixed.